addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1527,12 +1537,16 @@ var SnippetsMenu = Widget.extend({
                 // ... if no editors are to be enabled, look if any have been
                 // enabled previously by a click
                 if (!editorToEnable) {
-                     editorToEnable = this.snippetEditors.find(editor => editor.isSticky());
-                     previewMode = false;
+                    editorToEnable = this.snippetEditors.find(editor => editor.isSticky());
+                    previewMode = false;
                 }
                 // ... then enable the right editor
                 if (editorToEnable) {
                     editorToEnable.toggleOverlay(true, previewMode);
+                    if (!editorToEnable.displayHandles) {
+                        const parentEditor = editorToEnableHierarchy.find(ed => ed.displayHandles);
+                        parentEditor && parentEditor.toggleOverlay(true, previewMode);
```

Comment:
https://www.codereadability.com/dont-use-short-circuiting-when-you-want-an-if-statement/ :wink: You'll learn to know I am annoying with conventions and stuff like that :wink: